### PR TITLE
Added streak functionality

### LINKED ISSUE #55: UR-H #35: User should be able to view the streak score of a habit they own
Expected Behavior:
In the “Habit list Screen” or “Profile Screen”, the streak score of each habit that they own is shown as a number next to a fire emoji on a list of habits they own. 

Note: A streak score is the number of days the habit has been completed as scheduled in succession from the last missed occurrence. 

Mechanism:
The streak score of each habit along with the habit it is associated with will be loaded and cached from the database, showing a loading indicator while waiting for the read to occur

Edge Case Behavior:
Database entry contains invalid data -> Default values will be used

Database read -> Database read will be attempted again, if fails again the activity will be popped and the user will be displayed an error message.

Risk: Low
Story Points: 5

User Stories: US 01.09.01

### PR DESCRIPTION
**Pull request by loravocado**
## Screenshots

https://user-images.githubusercontent.com/41885935/143418268-3c90eecf-3fa4-427e-a4d4-e942b3e76e5a.mp4

## Checklist

- [ ] Automated tests
- [x] Screenshots included (if necessary)
- [ ] Code is well commented

Closes #55

### CODE REVIEW

**Comment by loravocado** [comment]
There's a bunch of tests broken not from my PR, but is blocking me from testing my tests, so I'll fix those all in another PR

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/DefineHabitEventActivity.java [hunk 1]
```diff
@@ -211,15 +219,23 @@ public boolean onOptionsItemSelected(MenuItem item) {
                     HabitEventModel habitEvent = new HabitEventModel(null,
                             locationSwitch.isChecked() && (markerLocation.getLatitude() != null) ? markerLocation : null,
                             new Date(), commentStr, null, habitID);
-
                     if (isNewHabitEvent) {
                         habitEventController.createHabitEvent(habitEvent, new HabitEventController.HabitEventIDCallback() {
                             @Override
                             public void onCallback(String habitEventID) {
                                 // TODO: figure out what to add here
```
Comment (AtrazineC) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/DefineHabitEventActivity.java:226]: is this TODO statement no longer needed?

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/WeeklyHabitScheduleModel.java [hunk 2]
```diff
@@ -62,12 +62,46 @@ public static WeeklyHabitScheduleModel parseList(List<String> list) {
...
+        if (date.after(today)) {
+            return days[testedDateOfWeek];
+        }
+        return false;
     }
 
     @Override
     public boolean wasSkippedIfLastCompletedOn(Date date) {
-        return false; // TODO
+        int lastCompletedDateOfWeek = date.getDay();
```
Comment (AtrazineC) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/WeeklyHabitScheduleModel.java:75]: a few minor styling issues in this function, try selecting the function contents and running Ctrl+Alt+L

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/DefineHabitEventActivity.java (hunk 1 above)
Comment (loravocado) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/DefineHabitEventActivity.java:226]: I didn't add it in the first place, so I'm not sure if whoever added it is done with it. I believe @MacMalainey added it

**Review by AtrazineC — DISMISSED** [review]
Looks awesome other than a few small nits! 💯

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/DefineHabitEventActivity.java (hunk 1 above)
Comment (MacMalainey) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/DefineHabitEventActivity.java:226]: Never have edited this file.

**Review by MacMalainey — DISMISSED** [review]

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/DefineHabitEventActivity.java (hunk 1 above)
Comment (j-vent) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/DefineHabitEventActivity.java:226]: That todo can be deleted

**Review by cheetiantey — APPROVED** [review]
LGTM!

**Review by AtrazineC — APPROVED** [review]

**Pull request merged**